### PR TITLE
feat: make form template settings easily accessible

### LINKED ISSUE #4563: feat: make form template settings easily accessible
# Feature Request

## Developer Story

As a developer, I want to make form template settings easily accessible.

Most of the form template required form template setting and shortcode argument to render the form. We do not have any system, in place to call that data easily.

## Note
- <s>When developer query for form setting then logic must handle default value and return when necessary.</s> No need to handle default values because they will be autofill when admin will create a new form. Admin can keep them or edit them. No need to have logic for it.

## Acceptance Criteria

- [x] Form template can access form settings and shortcode arguments easily.

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

This Pr will resolve #4563 

## How Has This Been Tested?
I tested this pr by processing donation with embed form.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: src/Helpers/Form/Utils.php [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 namespace Give\Helpers\Form\Utils;
 
-/**
- * Get result if we are viewing embed form or not
- *
- * @since 2.7.0
- *
- * @return bool
- */
+use WP_Post;/**
```
Comment (JasonTheAdams) [inline src/Helpers/Form/Utils.php]: Please correct this formatting.

File: src/Helpers/Form/Utils.php [hunk 2]
```diff
@@ -83,3 +82,46 @@ function getFailedTransactionPageURL( $args = array() ) {
 	);
 }
 
+/**
+ * Get shortcode argument.
```
Comment (JasonTheAdams) [inline src/Helpers/Form/Utils.php]: Please provide a description of _when_ this function ought to be used. What context is it assuming?

File: src/Helpers/Form/Utils.php [hunk 3]
```diff
@@ -83,3 +82,46 @@ function getFailedTransactionPageURL( $args = array() ) {
...
+		return false;
+	}
+
+	$queryString = array_map( 'give_clean', wp_parse_args( $_SERVER['QUERY_STRING'] ) );
+
+	return array_intersect_key( $queryString, give_get_default_form_shortcode_args() );
+}
+
+/**
+ * Get form ID.
```
Comment (JasonTheAdams) [inline src/Helpers/Form/Utils.php]: Same thing here. When would someone reach for this utility?

File: src/Views/Form/Themes/Sequoia/Actions.php [hunk 4]
```diff
@@ -3,6 +3,7 @@
 namespace Give\Views\Form\Themes\Sequoia;
 
 use Give_Donate_Form;
+use function Give\Helpers\Form\Theme\get;
```
Comment (JasonTheAdams) [inline src/Views/Form/Themes/Sequoia/Actions.php]: Please use an alias `Theme\get as getTheme`, as `get` is way too vague.

File: src/Helpers/Form/Theme.php [hunk 5]
```diff
@@ -6,17 +6,18 @@
...
  *
  * @param int $formID
  *
- * @return mixed
+ * @return string
  * @since 2.7.0
  */
-function getActiveID( $formID ) {
-	return Give()->form_meta->get_meta( $formID, '_give_form_theme', true );
+function getActiveID( $formID = null ) {
```
Comment (JasonTheAdams) [inline src/Helpers/Form/Theme.php]: Please document the `$formId` in the phpdoc and explain what the default does.

File: src/Form/LoadTheme.php [hunk 6]
```diff
@@ -40,26 +40,28 @@ class LoadTheme {
...
 	/**
-	 * Form Theme loading handler
+	 * setup form template
 	 *
-	 * @param string $formTheme Theme ID. Add form_theme shortcode argument to load selective form theme.
-	 *
-	 * @global WP_Post $post
+	 * @since 2.7.0
+	 * @param string $formTemplate
+	 * @param int    $formId
```
Comment (JasonTheAdams) [inline src/Form/LoadTheme.php]: Please document what this defaults to.

File: src/Helpers/Form/Utils.php (hunk 1 above)
Comment (ravinderk) [reply src/Helpers/Form/Utils.php]: fixed: https://github.com/impress-org/givewp/compare/de50ef46885f...349f67dcf3fc

File: src/Helpers/Form/Theme.php (hunk 5 above)
Comment (ravinderk) [reply src/Helpers/Form/Theme.php]: fixed: https://github.com/impress-org/givewp/compare/15fa2b13310e...b38638d7c2b3

File: src/Helpers/Form/Utils.php (hunk 3 above)
Comment (ravinderk) [reply src/Helpers/Form/Utils.php]: fixed: https://github.com/impress-org/givewp/compare/15fa2b13310e...b38638d7c2b3

File: src/Form/LoadTheme.php (hunk 6 above)
Comment (ravinderk) [reply src/Form/LoadTheme.php]: fixed:https://github.com/impress-org/givewp/compare/b38638d7c2b3...965ab69fb3f6

File: src/Views/Form/Themes/Sequoia/Actions.php (hunk 4 above)
Comment (ravinderk) [reply src/Views/Form/Themes/Sequoia/Actions.php]: fixed: https://github.com/impress-org/givewp/compare/965ab69fb3f6...fb342b64d1a7

File: src/Helpers/Form/Utils.php (hunk 2 above)
Comment (ravinderk) [reply src/Helpers/Form/Utils.php]: fixed: https://github.com/impress-org/givewp/compare/fb342b64d1a7...b03792348c28

**Comment by ravinderk** [comment]
@JasonTheAdams I update pr as per tour suggestions.

image

Above screenshot of updated `Helpers` directory.

File: src/Helpers/Form/Theme/Utils/Admin.php [hunk 7]
```diff
@@ -1,54 +1,11 @@
...
- * @param $settings
- *
- * @return mixed
- */
-function set( $formID, $settings ) {
-	$theme = Give()->form_meta->get_meta( $formID, '_give_form_theme', true );
-
-	return Give()->form_meta->update_meta( $formID, "_give_{$theme}_form_theme_settings", $settings );
-}
+use function Give\Helpers\Form\Theme\get;
```
Comment (JasonTheAdams) [inline src/Helpers/Form/Theme/Utils/Admin.php]: You did it again. Please avoid importing vague function names without providing an alias. 😉

Comment (ravinderk) [reply src/Helpers/Form/Theme/Utils/Admin.php]: fixed: https://github.com/impress-org/givewp/compare/ea7f3cc012f4...c9882bca1605

**Pull request merged**